### PR TITLE
Add option for comparing basecases when using the --compare-results o…

### PR DESCRIPTION
**Pull request by massimo-ferraro**
…ption in loadflow validation tool

### CODE REVIEW

File: loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ValidationTool.java [hunk 1]
```diff
@@ -139,7 +144,7 @@ public Command getCommand() {
     public void run(CommandLine line, ToolRunningContext context) throws Exception {
         Path caseFile = Paths.get(line.getOptionValue(CASE_FILE));
         Path outputFolder = Paths.get(line.getOptionValue(OUTPUT_FOLDER));
-        if (!Files.exists(outputFolder)) {
+        if (!outputFolder.toFile().exists()) {
```
Comment (mathbagu) [inline loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ValidationTool.java]: I’m sure you have to follow sonar advice for this smell, in particular if the Path has been created from a file system implementation that cannot give you a File (JimFS, remote file system...)

File: loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ValidationTool.java [hunk 2]
```diff
@@ -231,43 +235,39 @@ private void runValidation(Network network, ValidationConfig config, Set<Validat
...
-            });
-            writers.values().forEach(writer -> {
-                try {
-                    writer.close();
-                } catch (IOException e) {
-                    throw new UncheckedIOException(e);
-                }
-            });
+        public String toString() {
+            return this.name() + "(" + this.description + ")";
```
Comment (mathbagu) [inline loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ValidationTool.java]: Consider adding a space before the parenthesis

File: loadflow/loadflow-validation/src/test/java/com/powsybl/loadflow/validation/io/ValidationWritersTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                                                                 ValidationConfig.TABLE_FORMATTER_FACTORY_DEFAULT, ValidationConfig.EPSILON_X_DEFAULT,
+                                                                 ValidationConfig.APPLY_REACTANCE_CORRECTION_DEFAULT, ValidationOutputWriter.CSV_MULTILINE,
+                                                                 new LoadFlowParameters(), ValidationConfig.OK_MISSING_VALUES_DEFAULT,
+                                                                 ValidationConfig.NO_REQUIREMENT_IF_REACTIVE_BOUND_INVERSION_DEFAULT, ValidationConfig.COMPARE_RESULTS_DEFAULT,
+                                                                 ValidationConfig.CHECK_MAIN_COMPONENT_ONLY_DEFAULT, ValidationConfig.NO_REQUIREMENT_IF_SETPOINT_OUTSIDE_POWERS_BOUNDS);
+    private final Set<ValidationType> usedValidationTypes = Sets.immutableEnumSet(ValidationType.BUSES, ValidationType.FLOWS, ValidationType.GENERATORS);
+    private final Set<ValidationType> unusedValidationTypes = Sets.immutableEnumSet(ValidationType.SHUNTS, ValidationType.SVCS, ValidationType.TWTS);
+
+    FileSystem fileSystem;
+    ValidationWriters validationWriters;
```
Comment (mathbagu) [inline loadflow/loadflow-validation/src/test/java/com/powsybl/loadflow/validation/io/ValidationWritersTest.java]: Could it be private? Or must it be private package?

File: loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ValidationTool.java (hunk 1 above)
Comment (massimo-ferraro) [reply loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ValidationTool.java]: So your suggestion is, despite the sonarqube major issue, to use Files.exists, in order to possibly cover, and run the itool command in, the different file system implementations?

Comment (mathbagu) [reply loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ValidationTool.java]: Exactly. We are not fully inline with this sonar suggestion, so while we don't find the best thing to do, use Files.exists(Path)

**Pull request merged**